spec/lib/packet-spec.js

```diff
@@ -86,4 +87,150 @@ describe("Packet", function() {
+                    type: { value: 1, name: 'HW_ETHERNET' },
+                    address: '08:00:27:9b:d9:be'
+                },
+                sname: '',
+                fname: 'this is too long..................................................' +
+                       'this is too long..................................................'
+            };
+
+            expect(function() {
+                packetUtil.createPacketBuffer(pkt);
+            }).to.throw(Error);
+        });
+
+        it("should throw an error if the packet chaddr hw buffer does not equal 6 ':'", function() {
+            var pkt  = {
+                xid: 681302462,
+                chaddr: "123"
+            };
+
+            expect(function() {
+                packetUtil.createPacketBuffer(pkt);
+            }).to.throw(Error);
+        });
+
+        it("should thow an error if the packet chaddr is an object type with no 'address' key", function() {
+            var pkt  = {
+                xid: 681302462,
+                chaddr: {
+                    type: { value: 1, name: 'HW_ETHERNET' }
+                }
+            };
+
+            expect(function() {
+                packetUtil.createPacketBuffer(pkt);
+            }).to.throw(Error);
+        });
+
+        it("should parse the options", function() {
+            var pkt = {
+                xid: 681302462,
+                chaddr: {
+                    type: { value: 1, name: 'HW_ETHERNET' },
+                    address: '08:00:27:9b:d9:be'
+                },
+                options:
+                {
+                    subnetMask: '0.0.0.0',                  //option 1
+                    routerOptions: 'test string',           //option 3
+                    broadcastAddress: '255.255.255.255',    //option 28
+                    requestedIpAddress: '192.111.1.100',    //option 50
+                    ipAddressLeaseTime: 86400,              //option 51
+                    optionOverload: 100,                    //option 52
+                    dhcpMessageType: 0x2,                   //option 53
+                    serverIdentifier: '192.111.1.1',        //option 54
+                    parameterRequestList:                   //option 55
+                        [ 1, 2, 3, 4, 5, 6, 11, 12, 13, 15, 16, 17, 18, 22, 23, 28, 40, 41, 42, 43,
+                            50, 51, 54, 58, 59, 60, 66, 67, 128, 129, 130, 131, 132, 133, 134, 135 ],
+                    renewalTimeValue: 200,                  //option 58
+                    rebindingTimeValue: 300,                //option 59
+                    vendorClassIdentifier: 'PXEClient:Arch:00000:UNDI:002001',   //option 60
+                    clientIdentifier: '100.100', //option 61
+                    bootFileName: 'bootfilename', //option 67
+                    maximumMessageSize: 1260
+                }
+            };
+
+            var p = packetUtil.createPacketBuffer(pkt);
+            console.log(p);
```

Comment:
Also, do you want to assert anything about the packet buffer? `instanceof Buffer` or something?